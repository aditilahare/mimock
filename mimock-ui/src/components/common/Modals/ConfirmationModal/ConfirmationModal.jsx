import React from 'react';
import PropTypes from 'prop-types';
import { Portal } from 'react-portal';
import {
	ModalWrapper,
	ModalBox,
	ModalMessage,
	CancelButton,
	ConfirmButton,
} from './ConfirmationModal.style';

function ConfirmationModal({
	message,
	cancelButtonLabel,
	confirmButtonLabel,
	onConfirm,
	onCancel,
}) {
	return (
		<Portal node={document && document.getElementById('app')}>
			<ModalWrapper data-testid='confirmation-modal'>
				<ModalBox>
					<ModalMessage data-testid='confirmation-modal-message'>
						{message}
					</ModalMessage>
					<CancelButton
						data-testid='confirmation-modal-cancel-btn'
						onClick={onCancel}
					>
						{cancelButtonLabel}
					</CancelButton>
					<ConfirmButton
						data-testid='confirmation-modal-confirm-btn'
						onClick={onConfirm}
					>
						{confirmButtonLabel}
					</ConfirmButton>
				</ModalBox>
			</ModalWrapper>
		</Portal>
	);
}

ConfirmationModal.propTypes = {
	message: PropTypes.string.isRequired,
	cancelButtonLabel: PropTypes.string,
	confirmButtonLabel: PropTypes.string,
	onConfirm: PropTypes.func.isRequired,
	onCancel: PropTypes.func.isRequired,
};

ConfirmationModal.defaultProps = {
	message: 'Are you sure you want proceed with this action?',
	cancelButtonLabel: 'Cancel',
	confirmButtonLabel: 'Confirm',
};

export default ConfirmationModal;
