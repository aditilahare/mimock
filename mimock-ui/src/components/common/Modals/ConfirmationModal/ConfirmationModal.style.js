import tw from 'tailwind-styled-components';

export const ModalWrapper = tw.div`
    w-screen
    h-screen
    flex
    justify-center
    items-center
    bg-slate-500/[0.8]
    align-middle
    fixed
    left-0
    right-0
    top-0
    bottom-0
`;

export const ModalBox = tw.div`
    w-1/2
    rounded-md
    shadow-lg
    bg-white
    p-6
`;

export const ModalMessage = tw.div`
    text-center
    font-sans
    font-semibold
    text-xl
    my-4
    text-gray-700
`;

const _ModalButtons = tw.div`
    w-1/2
    text-center
    font-sans
    font-semibold
    text-lg
    mx-auto
    mt-8
    cursor-pointer
    transition-all
    duration-100
    hover:scale-95
`;

export const CancelButton = tw(_ModalButtons)`
    text-gray-500
    hover:text-gray-900
`;

export const ConfirmButton = tw(_ModalButtons)`
    bg-teal-500
    text-white
    rounded-full
    shadow-md
    p-4
    mb-4
    hover:bg-teal-600
`;
