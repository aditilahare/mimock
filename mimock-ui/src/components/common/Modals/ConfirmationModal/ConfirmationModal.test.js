import React from 'react';
import ConfirmationModal from './ConfirmationModal';
import { render, act, fireEvent } from '@testing-library/react';

describe('ConfirmationModal', () => {
	const mockCancel = jest.fn();
	const mockConfirm = jest.fn();
	const title = 'Are you sure you want to do this?';

	it('Should render confirmation modal with default props', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ConfirmationModal onConfirm={mockConfirm} onCancel={mockCancel} />
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		expect(getByTestId('confirmation-modal-message')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message').textContent).toBe(
			'Are you sure you want proceed with this action?'
		);

		expect(getByTestId('confirmation-modal-cancel-btn')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-cancel-btn').textContent).toBe(
			'Cancel'
		);

		expect(getByTestId('confirmation-modal-confirm-btn')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-confirm-btn').textContent).toBe(
			'Confirm'
		);

		expect(document.body).toMatchSnapshot();
	});

	it('Should render confirmation modal with passed props', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ConfirmationModal
					message={title}
					onConfirm={mockConfirm}
					onCancel={mockCancel}
					cancelButtonLabel={'No'}
					confirmButtonLabel={'Yes'}
				/>
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		expect(getByTestId('confirmation-modal-message')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message').textContent).toBe(title);

		expect(getByTestId('confirmation-modal-cancel-btn')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-cancel-btn').textContent).toBe('No');

		expect(getByTestId('confirmation-modal-confirm-btn')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-confirm-btn').textContent).toBe(
			'Yes'
		);

		expect(document.body).toMatchSnapshot();
	});

	it('Should trigger cancel event on clicking cancel button', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ConfirmationModal
					message={title}
					onConfirm={mockConfirm}
					onCancel={mockCancel}
					cancelButtonLabel={'No'}
					confirmButtonLabel={'Yes'}
				/>
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		expect(getByTestId('confirmation-modal-message')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message').textContent).toBe(title);

		const cancelButton = getByTestId('confirmation-modal-cancel-btn');
		expect(cancelButton).toBeInTheDocument();
		expect(cancelButton.textContent).toBe('No');
		await fireEvent.click(cancelButton);
		expect(mockCancel).toHaveBeenCalledTimes(1);

		expect(document.body).toMatchSnapshot();
	});

	it('Should trigger confirm event on clicking confirm button', async () => {
		let tree;
		await act(async () => {
			tree = await render(
				<ConfirmationModal
					message={title}
					onConfirm={mockConfirm}
					onCancel={mockCancel}
					cancelButtonLabel={'No'}
					confirmButtonLabel={'Yes'}
				/>
			);
		});

		const { getByTestId } = tree;

		expect(getByTestId('confirmation-modal')).toBeInTheDocument();

		expect(getByTestId('confirmation-modal-message')).toBeInTheDocument();
		expect(getByTestId('confirmation-modal-message').textContent).toBe(title);

		const confirmButton = getByTestId('confirmation-modal-confirm-btn');
		expect(confirmButton).toBeInTheDocument();
		expect(confirmButton.textContent).toBe('Yes');
		await fireEvent.click(confirmButton);
		expect(mockConfirm).toHaveBeenCalledTimes(1);

		expect(document.body).toMatchSnapshot();
	});
});
