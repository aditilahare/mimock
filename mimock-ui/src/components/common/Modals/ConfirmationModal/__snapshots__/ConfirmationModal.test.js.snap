// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal Should render confirmation modal with default props 1`] = `
<body>
  <div />
  <div>
    <div
      class="w-screen h-screen flex justify-center items-center bg-slate-500/[0.8] align-middle fixed left-0 right-0 top-0 bottom-0"
      data-testid="confirmation-modal"
    >
      <div
        class="w-1/2 rounded-md shadow-lg bg-white p-6"
      >
        <div
          class="text-center font-sans font-semibold text-xl my-4 text-gray-700"
          data-testid="confirmation-modal-message"
        >
          Are you sure you want proceed with this action?
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 text-gray-500 hover:text-gray-900"
          data-testid="confirmation-modal-cancel-btn"
        >
          Cancel
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 bg-teal-500 text-white rounded-full shadow-md p-4 mb-4 hover:bg-teal-600"
          data-testid="confirmation-modal-confirm-btn"
        >
          Confirm
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ConfirmationModal Should render confirmation modal with passed props 1`] = `
<body>
  <div />
  <div>
    <div
      class="w-screen h-screen flex justify-center items-center bg-slate-500/[0.8] align-middle fixed left-0 right-0 top-0 bottom-0"
      data-testid="confirmation-modal"
    >
      <div
        class="w-1/2 rounded-md shadow-lg bg-white p-6"
      >
        <div
          class="text-center font-sans font-semibold text-xl my-4 text-gray-700"
          data-testid="confirmation-modal-message"
        >
          Are you sure you want to do this?
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 text-gray-500 hover:text-gray-900"
          data-testid="confirmation-modal-cancel-btn"
        >
          No
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 bg-teal-500 text-white rounded-full shadow-md p-4 mb-4 hover:bg-teal-600"
          data-testid="confirmation-modal-confirm-btn"
        >
          Yes
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ConfirmationModal Should trigger cancel event on clicking cancel button 1`] = `
<body>
  <div />
  <div>
    <div
      class="w-screen h-screen flex justify-center items-center bg-slate-500/[0.8] align-middle fixed left-0 right-0 top-0 bottom-0"
      data-testid="confirmation-modal"
    >
      <div
        class="w-1/2 rounded-md shadow-lg bg-white p-6"
      >
        <div
          class="text-center font-sans font-semibold text-xl my-4 text-gray-700"
          data-testid="confirmation-modal-message"
        >
          Are you sure you want to do this?
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 text-gray-500 hover:text-gray-900"
          data-testid="confirmation-modal-cancel-btn"
        >
          No
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 bg-teal-500 text-white rounded-full shadow-md p-4 mb-4 hover:bg-teal-600"
          data-testid="confirmation-modal-confirm-btn"
        >
          Yes
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ConfirmationModal Should trigger confirm event on clicking confirm button 1`] = `
<body>
  <div />
  <div>
    <div
      class="w-screen h-screen flex justify-center items-center bg-slate-500/[0.8] align-middle fixed left-0 right-0 top-0 bottom-0"
      data-testid="confirmation-modal"
    >
      <div
        class="w-1/2 rounded-md shadow-lg bg-white p-6"
      >
        <div
          class="text-center font-sans font-semibold text-xl my-4 text-gray-700"
          data-testid="confirmation-modal-message"
        >
          Are you sure you want to do this?
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 text-gray-500 hover:text-gray-900"
          data-testid="confirmation-modal-cancel-btn"
        >
          No
        </div>
        <div
          class="w-1/2 text-center font-sans font-semibold text-lg mx-auto mt-8 cursor-pointer transition-all duration-100 hover:scale-95 bg-teal-500 text-white rounded-full shadow-md p-4 mb-4 hover:bg-teal-600"
          data-testid="confirmation-modal-confirm-btn"
        >
          Yes
        </div>
      </div>
    </div>
  </div>
</body>
`;
